src/Compilers/CSharp/Test/Semantic/Semantics/NullableReferenceTypesTests.cs

```diff
@@ -46358,6 +46358,32 @@ void f()
                 );
         }
 
+        [Fact]
+        [WorkItem(42396, "https://github.com/dotnet/roslyn/issues/42396")]
+        public void LocalFunction_05()
```

Comment:
I see that we also have logic involving `_delegateInvokeMethod` in `NullableWalker.TryGetReturnType()` (used in `VisitReturnStatement`). Could we add a similar test case for vanilla `return` statement in non-enumerable nested local function?

Reply:
Makes me wonder if we need `_delegateInvokeMethod` at all if it's not for such nested scenarios...

Reply:
We shouldn't except that lambdas are analyzed in a separate walker.